src/lib/tabs/tab-header.ts

```diff
@@ -0,0 +1,258 @@
+import {
+  ViewChild,
+  Component,
+  Input,
+  NgZone,
+  QueryList,
+  ElementRef,
+  Renderer, ViewEncapsulation, ContentChildren, Output, EventEmitter
+} from '@angular/core';
+import {RIGHT_ARROW, LEFT_ARROW, ENTER, Dir, LayoutDirection} from '../core';
+import {MdTabLabelWrapper} from './tab-label-wrapper';
+import {MdInkBar} from './ink-bar';
+import 'rxjs/add/operator/map';
+
+export type ScrollDirection = 'after' | 'before';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-tab-header',
+  templateUrl: 'tab-header.html',
+  styleUrls: ['tab-header.css'],
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    '[class.md-tab-header]': 'true',
+    '[class.scroll]': '_isScrollingEnabled()',
```

Comment:
I think that caching the width and only updating it in the following cases should be fine:
1. When initialized.
2. When a tab gets added/removed.
3. When a tab's heading changes.

The third one might be tricky to get right since we'd have to either get the `innerText` from the DOM on every change detection, or use a `MutationObserver` per tab. We could also try to have have one "main" `MutationObserver` on the header, but I don't know whether that won't end up firing too often, because AFAIK, you need to enable a bunch of flags to get it working.